controllers/nodehealthcheck_controller.go

```diff
@@ -196,7 +198,11 @@ func (r *NodeHealthCheckReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 
 	// always check if we need to patch status before we exit Reconcile
 	nhcOrig := nhc.DeepCopy()
+	var finalRequeueAfter *time.Duration
 	defer func() {
+		if finalRequeueAfter != nil {
+			result.RequeueAfter = *finalRequeueAfter
```

Comment:
Nit: I'd consider to check here for min as well - mainly to safeguard against future regression